azure/notebooks/azure_mnmg_xgboost.ipynb

```diff
@@ -0,0 +1,1287 @@
+    "# # Installing azureml-opendatasets with for some weird reason installs an old version of Pandas 1.0.0, which we will upgrade to 1.2.4 to work with RAPIDS. \n",
+    "# # We do the same for the numpy and scipy libraries that are also downgraded by azureml-opendatasets.\n",
+    "# ! pip install \"dask-cloudprovider[azure]\"\n",
+    "# ! pip install \"dask-cloudprovider[azure]\" --upgrade\n",
+    "# ! pip install azureml-core\n",
+    "# ! pip install azureml-opendatasets\n",
+    "# ! pip install azureml-telemetry\n",
+    "# ! pip install pandas==1.2.4 # reverting pandas to 1.2.4\n",
+    "# ! pip install numpy==1.20.2 # reverting numpy to 1.20.2\n",
+    "# ! pip install scipy==1.6.0 # reverting scipy to 1.16.0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import Client, wait, get_worker\n",
+    "from dask_cloudprovider.azure import AzureVMCluster\n",
+    "import dask_cudf\n",
+    "from azureml.opendatasets import NycTlcYellow\n",
+    "from dask_ml.model_selection import train_test_split\n",
+    "from cuml.dask.common import utils as dask_utils\n",
+    "from cuml.metrics import mean_squared_error\n",
+    "from cuml import ForestInference\n",
+    "import cudf\n",
+    "import xgboost as xgb\n",
+    "from datetime import datetime\n",
+    "from dateutil import parser\n",
+    "import numpy as np\n",
+    "from timeit import default_timer as timer\n",
+    "import dask\n",
+    "import json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## <span style=\"color:#8735fb; font-size:22pt\"> Step 1: Set up the Azure VM Cluster</span>\n",
+    "\n",
+    "We will now set up an [Azure VM Cluster](https://cloudprovider.dask.org/en/latest/azure.html) using `AzureVMCluster` from Dask Cloud Provider, which creates a cluster of VMs to run our workload. To do this, you will first need to set up a Resource Group, a Virtual Network and a Security Group on Azure. [Learn more about how you can set this up](https://cloudprovider.dask.org/en/latest/azure.html#resource-groups). Note that you can also set it up using the Azure portal directly via the GUI.\n",
+    "\n",
+    "Once you have set it up, you can now plug in the names of the entities you have created in the cell below. \n",
+    "\n",
+    "We need to pass in the docker argument `docker_args = '--shm-size=256m'` to allow larger shared memory for successfully running multiple docker containers in the same VM. This is the case when each VM has more than one worker. Even if you don't have such a case, there is no harm in having a larger shared memory. Finally, note that we use the RAPIDS docker image to build the VM and use the `dask_cuda.CUDAWorker` to run within the VM. This will run the worker docker image with cuda capabilities instead of cpu capabilities."
```

Comment:
```suggestion
    "We need to pass in the docker argument `docker_args = '--shm-size=256m'` to allow larger shared memory for successfully running multiple docker containers in the same VM. This is the case when each VM has more than one worker. Even if you don't have such a case, there is no harm in having a larger shared memory. Finally, note that we use the RAPIDS docker image to build the VM and use the `dask_cuda.CUDAWorker` to run within the VM. This will run the worker docker image with GPU capabilities instead of CPU."
```